File: src/main/kotlin/com/app/employeetddexample/otp/model/OtpGenerator.java
package com.app.employeetddexample.otp.model;

import java.util.Random;

 class OtpGenerator {

    char[] otpGen ( int len){
        String numbers = "0123456789";

        Random rndm_method = new Random();

        char[] otp = new char[len];

        for (int i = 0; i < len; i++) {
            otp[i] =
                    numbers.charAt(rndm_method.nextInt(numbers.length()));
        }
        return otp;
    }
     String generateRandomString(){
        int len =6;

        return  String.valueOf(otpGen(len));
    }
}

File: src/main/kotlin/com/app/employeetddexample/otp/model/OtpGenerator.kt
package com.app.employeetddexample.otp.model

import java.util.*

class OtpGenerator {
    private val CHAR_LIST = "1234567890"
    private val RANDOM_STRING_LENGTH = 6


    fun generateRandomString(): String {

        val randStr = StringBuffer()
        for (i in 0 until RANDOM_STRING_LENGTH) {
            val number = getRandomNumber()
            val ch = CHAR_LIST[number]
            randStr.append(ch)
        }
        return randStr.toString()
    }

    /**
     * This method generates random numbers
     * @return int
     */
    private fun getRandomNumber(): Int {
        var randomInt = 0
        val randomGenerator = Random()
        randomInt = randomGenerator.nextInt(CHAR_LIST.length)
        return if (randomInt - 1 == -1) {
            randomInt
        } else {
            randomInt - 1
        }

    }
}